### PR TITLE
Fix the join condition hash symbols ordering in the HashJoin.

### PR DESCRIPTION
**Pull request by kovrus**
## Summary of the changes / Why this improves CrateDB

It could have happened that joining more than 2 tables using hash join
won't return the joined tuples despite that the join condition is satisfied.
That could happen due to the non-deterministic ordering of the join condition
symbols (hash join symbols) which would lead to the wrong calculation of the
hash code for the join condition values on either join side. This change fixes
the ordering of the join condition symbols.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by kovrus** [comment]
i am not sure about an integration test for it. do you think it worth adding?

File: server/src/main/java/io/crate/planner/operators/HashJoinConditionSymbolsExtractor.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public Void visitFunction(Function function, Context context) {
                     int duplicatePos = 0;
                     for (Symbol arg : function.arguments()) {
                         var relation = arg.accept(RELATION_EXTRACTOR, null);
-                        List<Symbol> symbols = context.symbolsPerRelation.computeIfAbsent(relation, k -> new ArrayList<>());
+                        List<Symbol> symbols = context.symbolsPerRelation.computeIfAbsent(relation, k -> new LinkedList<>());
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/planner/operators/HashJoinConditionSymbolsExtractor.java:95]: Is this change necessary? I don't understand how this helps, `ArrayList` also keeps the insertion order

**Review by mfussenegger — COMMENTED** [review]
> i am not sure about an integration test for it. do you think it worth adding?

I think the unit tests are sufficient.


Why the backport to 4.1?

File: server/src/main/java/io/crate/planner/operators/HashJoinConditionSymbolsExtractor.java (hunk 1 above)
Comment (kovrus) [reply server/src/main/java/io/crate/planner/operators/HashJoinConditionSymbolsExtractor.java:95]: it slipped in, it is not needed.

**Review by mfussenegger — APPROVED** [review]
:+1:
